frontend/lib/routes/company/packages/AddPackageForm.dart

```diff
@@ -0,0 +1,358 @@
+        return;
+      }
+
+      Package? p = await _packageService.createPackage(
+          name: name, price: price, vat: vat, items: itemsPack);
+
+      if (p != null) {
+        PackageNotifier notifier =
+            Provider.of<PackageNotifier>(context, listen: false);
+        notifier.add(p);
+
+        Event? e = await _eventService.addPackageToEvent(
+            publicName: publicName, template: p.id);
+
+        if (e == null) {
+          ScaffoldMessenger.of(context).hideCurrentSnackBar();
+
+          ScaffoldMessenger.of(context).showSnackBar(
+            const SnackBar(
+                content: Text('Error: package not added to current event.',
+                    style: TextStyle(color: Colors.white))),
+          );
+        } else {
+          EventNotifier eventNotifier =
+              Provider.of<EventNotifier>(context, listen: false);
+
+          eventNotifier.event = e;
+
+          ScaffoldMessenger.of(context).hideCurrentSnackBar();
+
+          ScaffoldMessenger.of(context).showSnackBar(
+            const SnackBar(
+              content: Text('Done', style: TextStyle(color: Colors.white)),
+              duration: Duration(seconds: 2),
+            ),
+          );
+        }
+      } else {
+        ScaffoldMessenger.of(context).hideCurrentSnackBar();
+
+        ScaffoldMessenger.of(context).showSnackBar(
+          const SnackBar(
+              content: Text('An error occured.',
+                  style: TextStyle(color: Colors.white))),
+        );
+      }
+      Navigator.pop(context);
+    }
+  }
+
+  List<Item> getItems() {
+    ItemsNotifier notifier = Provider.of<ItemsNotifier>(context);
+    return notifier.getItems();
+  }
+
+  Widget _buildForm() {
+    return Form(
+      key: _formKey,
+      child: Column(
+        children: [
+          Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: TextFormField(
+              controller: _nameController,
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter a name';
+                }
+                return null;
+              },
+              decoration: const InputDecoration(
+                icon: const Icon(Icons.title),
+                labelText: "Package private name *",
+              ),
+            ),
+          ),
+          Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: TextFormField(
+              controller: _publicNameController,
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter a name';
+                }
+                return null;
+              },
+              decoration: const InputDecoration(
+                icon: const Icon(Icons.title),
+                labelText: "Package Public Name *",
+              ),
+            ),
+          ),
+          Row(
+            children: [
+              Expanded(
```

Comment:
Maybe we should have a field where people can insert any amount (e.g. 12.50) and then convert it internally to only cents for example?